js/typeahead-filter.js

```diff
@@ -167,43 +174,67 @@ TypeaheadFilter.prototype.appendCheckbox = function(opts) {
 
 TypeaheadFilter.prototype.getFilters = function(values) {
   var self = this;
-  var dataset = this.dataset.name + 's';
-  var idKey = this.dataset.name + '_id';
-  var ids = values.filter(function(value) {
-    return ID_PATTERN.test(value);
-  });
-  values.forEach(function(value) {
-    self.appendCheckbox({
-      name: self.fieldName,
-      id: slugify(value) + '-checkbox',
-      value: value,
-      label: ID_PATTERN.test(value) ? 'Loading...' : value
+  if (this.dataset) {
+    var dataset = this.dataset.name + 's';
+    var idKey = this.dataset.name + '_id';
+    var ids = values.filter(function(value) {
+      return ID_PATTERN.test(value);
     });
-  });
-  if (ids.length) {
-    $.getJSON(
-      URI(API_LOCATION)
-        .path([API_VERSION, dataset].join('/'))
-        .addQuery('api_key', API_KEY)
-        .addQuery(idKey, ids)
-        .toString()
-    ).done(this.updateFilters.bind(this));
+    values.forEach(function(value) {
+      self.appendCheckbox({
+        name: self.fieldName,
+        id: slugify(value) + '-checkbox',
+        value: value,
+        label: ID_PATTERN.test(value) ? 'Loading...' : value
+      });
+    });
+    if (ids.length) {
+      $.getJSON(
+        URI(API_LOCATION)
+          .path([API_VERSION, dataset].join('/'))
+          .addQuery('api_key', API_KEY)
+          .addQuery(idKey, ids)
+          .toString()
+      ).done(this.updateFilters.bind(this));
+    }
   }
 };
 
 TypeaheadFilter.prototype.updateFilters = function(response) {
   var self = this;
-  var idKey = this.dataset.name + '_id';
-  response.results.forEach(function(result) {
-    var label = result.name + ' (' + result[idKey] + ')';
-    self.$body.find('label[for="' + result[idKey] + '-checkbox"]').text(label);
-    self.$body.find('#' + result[idKey] + '-checkbox').trigger('filter:renamed', [
-      {
-        key: result[idKey] + '-checkbox',
-        value: label
-      }
-    ]);
-  });
+  if (this.dataset) {
+    var idKey = this.dataset.name + '_id';
+    response.results.forEach(function(result) {
+      var label = result.name + ' (' + result[idKey] + ')';
+      self.$body.find('label[for="' + result[idKey] + '-checkbox"]').text(label);
+      self.$body.find('#' + result[idKey] + '-checkbox').trigger('filter:renamed', [
+        {
+          key: result[idKey] + '-checkbox',
+          value: label
+        }
+      ]);
+    });
+  }
+};
+
+TypeaheadFilter.prototype.changeDataset = function(e, opts) {
+  // Method for changing the typeahead suggestion on the "contributor name" filter
+  // when the "individual" or "committee" checkbox filter is changed
+  // If the modify event names this filter, destroy it
+  if (opts.filterName === this.fieldName) {
+    this.$field.typeahead('destroy');
+    // If the value array is only individuals and not committees
+    // set the dataset to empty and re-init
+    if (opts.filterValue.indexOf('individual') > -1 && opts.filterValue.indexOf('committee') < 0) {
```

Comment:
If we were to abstract this out further for a more general use-case I'd be worried about accounting for an arbitrary number of checkboxes here.  We'd need another way to manage that in a more scalable and flexible manner.